include/aws/common/math.fallback.inl

```diff
@@ -66,3 +66,51 @@ AWS_STATIC_IMPL int aws_mul_u32_checked(uint32_t a, uint32_t b, uint32_t *r) {
     }
     return 1;
 }
+
+/**
+ * Adds a + b.  If the result overflows returns 2^64 - 1.
+ */
+AWS_STATIC_IMPL uint64_t aws_add_u64_saturating(uint64_t a, uint64_t b) {
+  uint64_t x = a + b;
```

Comment:
Another way to do this is to pre-check -> this requires one extra arithmetic operation but avoids ever having overflow behaviour
```
 if(UINT64_MAX - b < a) {
    return UINT64_MAX;
  }
  return a + b;
```